satpy/readers/clavrx.py

```diff
@@ -126,24 +140,21 @@ def _get_data(data, dataset_id: dict) -> xr.DataArray:
         factor = attrs.pop('scale_factor', (np.ones(1, dtype=data.dtype))[0])
         offset = attrs.pop('add_offset', (np.zeros(1, dtype=data.dtype))[0])
         valid_range = attrs.get('valid_range', [None])
-        if isinstance(valid_range, np.ndarray):
-            attrs["valid_range"] = valid_range.tolist()
 
         flags = not data.attrs.get("SCALED", 1) and any(data.attrs.get("flag_values", [None]))
         if not flags:
             data = data.where(data != fill)
             data = _CLAVRxHelper._scale_data(data, factor, offset)
             # don't need _FillValue if it has been applied.
             attrs.pop('_FillValue', None)
-
-        if all(valid_range):
-            valid_min = _CLAVRxHelper._scale_data(valid_range[0], factor, offset)
-            valid_max = _CLAVRxHelper._scale_data(valid_range[1], factor, offset)
-            if flags:
-                data = data.where((data >= valid_min) & (data <= valid_max), fill)
-            else:
+            if isinstance(valid_range, np.ndarray):
+                valid_min = _CLAVRxHelper._scale_data(valid_range[0], factor, offset)
```

Comment:
What is the proper way to patch this _scale_data function and use an assert_called() test?

Reply:
Easiest might be to extract `_scale_data` outside the class and then do the mocking. To do it "proper" (read: the way I would prefer) do:

```
from satpy.readers.clavrx import _scale_data
with mock.patch("satpy.readers.clavrx._scale_data", wraps=_scale_data) as scale_data:
    ...
scale_data.assert_called()
```

But overall it may be even easier if you just check the final result and make sure it has the scaled values you expect.